subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -71,9 +89,192 @@ static int cmd_send(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_exec(const struct shell *sh, size_t argc, char **argv) 
+{
+	struct lwm2m_ctx *ctx = lwm2m_rd_client_ctx();
+
+	if (!ctx) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	int ignore_cnt = 2;	/* Subcmd + PATH */
+	const char* pathstr = argv[1];
+	struct lwm2m_engine_res *res = NULL;
+	int ret = lwm2m_engine_get_resource(pathstr, &res);
+	if (ret < 0)
+	{
+		shell_error(sh, "Couldn't execute resource\n");
+		return ret;
```

Comment:
I believe you should not return just any error code. See https://docs.zephyrproject.org/latest/doxygen/html/group__shell__api.html#ga2546bd682966f71f03985f220ff48dc2

So maybe just print out the error code, and then return either `-ENOEXEC` or `-EINVAL`
```suggestion
		shell_error(sh, "Resource not found (err %d)\n", ret);
		return -EINVAL;
```